GitHub desktop port

usersxx opened the pull request:
#### Description

<!-- Note: it is best to make pull requests from a branch rather than from master -->

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [x] enhancement

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'printf "%s\n" "macOS `sw_vers -productVersion` `sw_vers -buildVersion` `uname -m`" "`xcodebuild -version|awk '\''NR==1{x=$0}END{print x" "$NF}'\''`"'|tee /dev/tty|pbcopy
-->
macOS x.y
Xcode x.y

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

i0ntempest reviewed (changes requested):
App bundles installed by MacPorts should be installed to `${applications_dir}` so you don't need `violate_mtree`. `extract.dir` should not be touched, rather you should write a `destroot` block to copy the app to `${destroot}${applications_dir}`. See qBittorrent portfile for an example.

i0ntempest commented:
`applications_dir` is set by MacPorts and you're not supposed to overwrite it. Also why are you copying the whole `${worksrcpath}`?

usersxx commented:
> App bundles installed by MacPorts should be installed to `${applications_dir}` so you don't need `violate_mtree`. `extract.dir` should not be touched, rather you should write a `destroot` block to copy the app to `${destroot}${applications_dir}`. See qBittorrent portfile for an example.

Removed `violate_mtree`, `extract.dir` in lines 37 and 34. Added `${applications_dir}`, `${destroot}${applications_dir}` in lines 34 and 36

usersxx commented:
The worksrcdir is set to `github desktop.app`

i0ntempest commented:
My bad for missing that, but variables set by MacPorts shouldn't be changed unless absolutely necessary. The right thing to do here is extract the app to the default `${worksrcpath}` and copy `${worksrcpath}${app_name}` to `${destroot}${applications_dir}`. By changing `${applications_dir}` you're still violating mtree.

i0ntempest commented:
Right, so I got your portfile and tried it. You indeed have to overwrite `worksrcdir`, but you should not dictate where to install an app for everyone in your portfile. `${applications_dir}` defaults to `/Applications/MacPorts/` and can be changed by a user in a config file. So `copy ${worksrcpath} ${destroot}${applications_dir}` will copy the app just fine.

Other things:
Version should be 2.9.6-9196a1ae, right now you're splitting it;
Change `extract.suffix      .zip` to `use_zip yes` so extraction actually works;
Your file name should be `Portfile` with a capital P. macOS filesystems can be case sensitive and they are on MacPorts buildbots. Pay attention to other places where you might mess up cases as well (Github vs GitHub etc.);
Maintainer field should look like this: `{@name domain:username}`, so for you maybe {@usersxx gmail.com:usersxx}.

i0ntempest commented on devel/Github-Desktop/Portfile:
```diff
@@ -3,17 +3,17 @@
 supported_archs     x86_64 arm64
 license             MIT 
-maintainers         openmaintainer usersxx
+maintainers         openmaintainer usersxx {@usersxx microsoft.com:usersword123}
```
change this line to `maintainers        {@usersxx microsoft.com:usersword123} openmaintainer`

i0ntempest commented on devel/Github-Desktop/Portfile:
```diff
@@ -0,0 +1,39 @@
+destroot {
+    copy ${worksrcpath} ${destroot}${applications_dir}
+}
+use_configure    no
```
Add a few whitespaces to format this line like other lines

usersxx commented:
> App bundles installed by MacPorts should be installed to `${applications_dir}` so you don't need `violate_mtree`. `extract.dir` should not be touched, rather you should write a `destroot` block to copy the app to `${destroot}${applications_dir}`. See qBittorrent portfile for an example.

You should approve changes

i0ntempest commented:
> > App bundles installed by MacPorts should be installed to `${applications_dir}` so you don't need `violate_mtree`. `extract.dir` should not be touched, rather you should write a `destroot` block to copy the app to `${destroot}${applications_dir}`. See qBittorrent portfile for an example.
> 
> You should approve changes so the changes requested section should go away

Not yet, after you've done with the changes you need to squash your commits into one and change your commit message to follow our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages). Basically:
`port_name: summary of changes`

usersxx commented:
Ok

i0ntempest commented:
Right, I did the format change for you and the checks are running now. You can now squash all changes and I'll merge if no other members have suggestions in a few days

usersxx commented:
> Right, I did the format change for you and the checks are running now. You can now squash all changes and I'll merge if no other members have suggestions in a few days

Ok

reneeotten reviewed (changes requested):
this PR should not be merged in my opinion. We want to build packages from source and not just copy binaries from a website - in cases like this just install the thing directly from upstream.

If you people really, really want to do this then you will need to make sure to disable automatic updates through the app since now MacPorts will have to take care of this, amongst probably other things.

i0ntempest commented:
> this PR should not be merged in my opinion. We want to build packages from source and not just copy binaries from a website - in cases like this just install the thing directly from upstream.
> 
> If you people really, really want to do this then you will need to make sure to disable automatic updates through the app since now MacPorts will have to take care of this, amongst probably other things.

Actually I've thought about that. This app is written in typescript, is there any port currently builds typescript to binary? If so then maybe it's doable... But it's probably too hard for a first time contributer to do though.

usersxx commented:
Actually there is a repository for github desktop. [https://github.com/desktop/desktop](url)

ryandesign commented on devel/Github-Desktop/Portfile:
```diff
@@ -0,0 +1,41 @@
+
+PortSystem          1.0
+
+name                Github-Desktop
```
Capitalize GitHub correctly.
```suggestion
name                GitHub-Desktop
```

ryandesign replied:
The name of the directory that the Portfile is in should be updated to match this capitalization.

ryandesign reviewed (commented):
I agree the port should build from source, if possible.

If not, I left some comments about improvements that can be made.

The port fails to activate for me on Catalina due to [this MacPorts bug](https://trac.macports.org/ticket/60749)—did it activate for you? On what macOS version? (You didn't fill in that part of the pull request template.)

```
Error: Failed to activate GitHub-Desktop: Image error: Source file /opt/local/var/macports/software/GitHub-Desktop/mpextract5jq2fMoo/Applications/MacPorts/GitHub Desktop.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Resources/._MainMenu.nib does not appear to exist (cannot lstat it).  Unable to activate port GitHub-Desktop.
```

usersxx commented on devel/Github-Desktop/Portfile:
```diff
@@ -0,0 +1,41 @@
+supported_archs     x86_64 arm64
+license             MIT 
+maintainers         usersxx {@usersxx microsoft.com:usersword123} openmaintainer 
+description         Github Desktop is the desktop version of Github
```
The pull request name is GitHub but the directory of the Portfile is macports-ports/devel/Github-Desktop/Portfile

usersxx commented:
I think you should see this file - https://github.com/desktop/desktop/blob/development/package.json
